CMakeLists.txt

```diff
@@ -121,6 +121,7 @@ include(CheckCXXCompilerFlag)
 set(COMMON_CXX_FLAGS
   -pipe
   -Wall
+  -fopenmp 
```

Comment:
To do it in a portable way, we should use use `find_package` and use the populated variable.

An example is shown [here](https://stackoverflow.com/a/51448364) :
```
cmake_minimum_required(VERSION 3.9)
project(solver LANGUAGES CXX)

find_package(OpenMP REQUIRED)
add_executable(solver solver.cc)
target_link_libraries(solver PRIVATE OpenMP::OpenMP_CXX)
```

Note that we may need to bump our minimum version for CMake, because OpenMP support got reworked in 3.9 (which is only one year old: I don't know if it's too recent to be easily used from most of Linux distribution).

Reply:
An alternative way:
```
find_package(OpenMP)
if (OPENMP_FOUND)
    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
    set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_EXE_LINKER_FLAGS}")
endif()
```
If OpenMP is not found, its acceleration #pragma will be disabled.

Reply:
This is doesn't always work (hence the revamp in 3.9), see https://cmake.org/Bug/view.php?id=15393